### ISSUE TITLE
Python: implement GlueCatalog._commit_table

### ISSUE THREAD

**Issue by antonysouthworth-halter** [apache/iceberg#8514]
### Feature Request / Improvement

This would allow us to update tables from Python, specifically I want to be able to update the `sortOrder` of my table.

Apologies if this is already in some roadmap document somewhere, feel free to close if having the GitHub issue is not helpful.

### Query engine

None

**Comment by Fokko** [comment]
@antonysouthworth-halter This is not implemented today but is high on the list. Also, other users are looking for snapshot expiration, which is a very similar operation. I've created [this](https://github.com/apache/iceberg/issues/8518) ticket which has to happen first, then updating the `SortOrder` should be straightforward. I'll keep this ticket open for visibility.

**Comment by Fokko** [comment]
Closing in favor of https://github.com/apache/iceberg-python/issues/29 (migrated to a separate repository).
